android/src/main/java/com/mattermost/networkclient/APIClientModule.kt

```diff
@@ -4,99 +4,124 @@ import com.facebook.react.bridge.*
 import okhttp3.OkHttpClient
 import okhttp3.Request
 import okhttp3.RequestBody.Companion.toRequestBody
-import com.facebook.react.bridge.Arguments;
+import okio.IOException
 
 class APIClientModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {
 
-    var client = OkHttpClient();
+    var sessionsClient = mutableMapOf<String, OkHttpClient.Builder>();
     var sessions = mutableMapOf<String, Request.Builder>()
 
     override fun getName(): String {
         return "APIClient"
     }
 
     @ReactMethod
-    fun createClientFor(baseUrl: String, config: ReadableMap, promise: Promise){
+    fun createClientFor(baseUrl: String, options: ReadableMap, promise: Promise) {
         try {
+            // Create the client and request builder
+            sessionsClient[baseUrl] = OkHttpClient().newBuilder();
```

Comment:
don't we need to check if the session already exists? or are we ok to override it?

Reply:
I think we're okay to override it - on the JS side; when `createClient` is called, I think the assumption is the client doesn't exist already; or maybe you want to create one again - even if there is one on the native side.